### ISSUE TITLE
Problems with EfficientNet v2 b0 inference in tf lite format

### ISSUE THREAD

**Issue by feff2** [tensorflow/tensorflow#79422]
### 1. System information

- OS Platform and Distribution: Windows 11
- TensorFlow installation : pip 
- TensorFlow library: 2.18.0

### 2. Code

import tensorflow as tf

input_shape = (224, 224, 3)
inputs = tf.keras.Input(shape=input_shape)

model = tf.keras.applications.EfficientNetv2b0(
    include_top=True,
    weights='imagenet',
    input_tensor=inputs
)

full_model = tf.keras.Model(inputs=inputs, outputs=model.outputs)

model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
model.summary()

converter = tf.lite.TFLiteConverter.from_keras_model(full_model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

tf.lite.experimental.Analyzer.analyze(model_content=tflite_model, gpu_compatibility=True)

with open('EfficientNet_float32.tflite', 'wb') as f:
    f.write(tflite_model)

print("TFLite model saved successfully.")

- When I run inference in python, the model outputs the correct solutions. But model is much slower than in the onnx format . Results for TF lite: 0.145 for onnx: 0.0108   CPU: 6-core AMD Risen 5 7500F
- When I run inference in Kotlin, the model outputs are incorrect. Model always gives the same wrong classes with less than 0.05 probs like nematode, microphone, matchstick . Also i can`t run inference on GPU or with using NNAPI on mobile phone. 
I switched one of default model in this example from tf repo: https://github.com/tensorflow/examples/tree/master/lite/examples/image_classification/android

I used tf.lite.experimental.Analyzer.analyze(model_content=tflite_model, gpu_compatibility=True) and got it:  Your model looks compatible with GPU delegate on TFLite runtime version 2.18.0.
This does not guarantee that your model will work well with GPU delegate because there could still be runtime incompatibililties.

**Comment by gaikwadrahul8** [comment]
Hi, @feff2 

Thank you for bringing this issue to our attention, I see this PR https://github.com/tensorflow/tensorflow/pull/74830 was merged on September 12, 2024 for Deprecating GPU compatibility experimental feature `from both tf/compiler/mlir/lite:flatbuffer_export and tf/lite/python/analyzer_wrapper:model_analyzer` but still we are getting below messages 

```
Your model looks compatible with GPU delegate on TFLite runtime version 2.18.0-rc0.
This does not guarantee that your model will work well with GPU delegate because there could still be runtime incompatibililties.
```
You mentioned that the inference time for the TFLite model is significantly slower (0.145 seconds) compared to the ONNX model (0.0108 seconds). This could be due to several factors like model optimization so ensure that you are using optimizations effectively. You can try using quantization techniques to reduce the model size and improve inference speed. Please refer this [official documentation](https://ai.google.dev/edge/litert/models/model_optimization) and also check if your CPU is being fully utilized. Sometimes the model may not be optimized for the specific hardware you are using.

The model outputs incorrect classes with low probabilities when run in Kotlin. This could be due to input preprocessing so ensure that the input images are preprocessed in the same way as during training. This includes resizing, normalization and any other transformations.

If possible could you please help us with your Github repo replicate similar behavior along with complete steps which will be great to investigate this issue further ?

Thank you for your cooperation and patience.

**Comment by feff2** [comment]
Hi @gaikwadrahul8 
I use this example: https://github.com/tensorflow/examples/tree/master/lite/examples/image_classification/android
I changed ImageClassifierHelper
[ImageClassifierHelper.zip](https://github.com/user-attachments/files/17677754/ImageClassifierHelper.zip)
I use the model that was created by the following code:
import tensorflow as tf

input_shape = (224, 224, 3)
inputs = tf.keras.Input(shape=input_shape)

model = tf.keras.applications.EfficientNetv2b0(
include_top=True,
weights='imagenet',
input_tensor=inputs
)

full_model = tf.keras.Model(inputs=inputs, outputs=model.outputs)

model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
model.summary()

converter = tf.lite.TFLiteConverter.from_keras_model(full_model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model = converter.convert()

tf.lite.experimental.Analyzer.analyze(model_content=tflite_model, gpu_compatibility=True)

with open('EfficientNet_float32.tflite', 'wb') as f:
f.write(tflite_model)

print("TFLite model saved successfully.")
and  added metadata according to this documentation: https://ai.google.dev/edge/litert/models/metadata?hl=en

**Comment by gaikwadrahul8** [comment]
Hi, @feff2

I'm able to create model successfully with above provided code snippet for reference here is [gist-file](https://colab.research.google.com/gist/gaikwadrahul8/88a1e4f2dc4ad41c6e945e171ee15c1b/tflite-issue-79422.ipynb), if possible could you please help us with yout Github repo (along with complete steps) to replicate similar behavior from our end which will help us to investigate this issue further ?

Thank you for your cooperation and patience.
